### ISSUE TITLE
Also detect version with `importlib.metadata.version`

### ISSUE THREAD

**Issue by Ravencentric** [BrianPugh/cyclopts#32]
The docs state [two ways cyclopts tries to determine your CLI app's version](https://cyclopts.readthedocs.io/en/latest/version.html)

Another way to determine this would be using https://docs.python.org/3.12/library/importlib.metadata.html#distribution-versions

```py
from importlib import metadata
metadata.version("package-name")
```

Also has a backport to support older python https://importlib-metadata.readthedocs.io/en/latest/

**Comment by BrianPugh** [comment]
I can add this! Should this have higher or lower priority over directly checking `mypackage.__version__`? My gut says higher priority. Or would directly checking `mypackage.__version__` be redundant (is it already handled by importlib)?

**Comment by Ravencentric** [comment]
Yes, importlib should be at the highest priority because it's the "official" way.

Reference:
https://peps.python.org/pep-0396/#pep-rejection

**Comment by BrianPugh** [comment]
This has been and released in v1.2.0.
